### PR TITLE
Improve revoke device page

### PR DESCRIPTION
**Pull request by nathunsmitty**
A handful of minor things:

- Adds a `<ProgressIndicator />` component while the client waits to receive a list of all endangered TLFs.
- Adds a storybook component for the revoke device page.
- Some general refactoring with props/styles/types.
- Makes endangered TLF list text selectable.

@akalin-keybase's original ticket also mentioned potentially making the list of endangered TLFs links. I think this may be best implemented with a `<Files />` component embedded within the page, e.g.,
screen shot 2018-07-10 at 6 11 13 pm
I didn't implement this feature in this PR because it looked like it would involve some refactor of the files component if we did it that way and I wanted to get some feedback first.

I also want to note that I couldn't get any endangered TLFs to occur when I tried to reproduce a situation in which they would occur. This could be due to an error on my part, though.

Currently, we don't show any type of special warning or list of files when a user revokes their current and only device. I'm probably going to file a ticket for that.

### CODE REVIEW

File: shared/devices/device-page/index.js [hunk 1]
```diff
@@ -76,9 +77,15 @@ const Render = (props: Props) => {
     metaOne = <Meta title="revoked" style={styles.meta} backgroundColor={globalColors.red} />
   }
 
+  const icon: IconType = {
+    backup: 'icon-paper-key-64',
+    desktop: 'icon-computer-64',
+    mobile: 'icon-phone-64',
```
Comment (songgao) [inline shared/devices/device-page/index.js]: Are there no 48 versions of these like the revoke ones?

File: shared/devices/device-revoke/index.js [hunk 2]
```diff
@@ -71,33 +101,34 @@ const DeviceRevoke = (props: Props) => (
...
+    alignItems: 'flex-start',
+    alignSelf: 'center',
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: globalColors.black_05,
+    borderRadius: 4,
+    height: 162,
+    marginBottom: globalMargins.small,
+    marginTop: globalMargins.small,
+    // overflowY: 'scroll',
```
Comment (songgao) [inline shared/devices/device-revoke/index.js]: Did you mean to remove this line?

Comment (nathunsmitty) [reply shared/devices/device-revoke/index.js]: Nope! Good catch!

File: shared/devices/device-page/index.js (hunk 1 above)
Comment (chrisnojima) [reply shared/devices/device-page/index.js]: in zeplin its 64 on both

File: shared/devices/device-revoke/index.js [hunk 3]
```diff
@@ -1,31 +1,49 @@
...
+      <Icon type={icon} />
+      <Text type="BodySemibold" style={styles.name}>
+        {name}
+      </Text>
+    </Box>
+  )
+}
 
-const Body = ({endangeredTLFs, name, currentDevice}) => (
+const Body = ({
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: we're moving towards not destructuring these props as parameters, it just makes the file take a lot of vertical space for very little
instead
```
const Body  = (props: Props) => (
... this.props.waiting? ...
)
```


or if you use a prop a bunch (usually unlikely) you can just destructure that one prop
```
const Body = (props: Props) => {
  const {waiting} = props
...
}
```

File: shared/devices/device-revoke/index.js [hunk 4]
```diff
@@ -35,34 +53,46 @@ const Body = ({endangeredTLFs, name, currentDevice}) => (
...
     )}
   </Box>
 )
 
 const DeviceRevoke = (props: Props) => (
   <Confirm
     body={
-      <Body endangeredTLFs={props.endangeredTLFs} name={props.name} currentDevice={props.currentDevice} />
+      <Body
+        // endangeredTLFs={props.endangeredTLFs}
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: this has to go back

File: shared/devices/device-revoke/index.js [hunk 5]
```diff
@@ -35,34 +53,46 @@ const Body = ({endangeredTLFs, name, currentDevice}) => (
...
+        endangeredTLFs={['nathunsmitty']}
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: 🔥 🔥 🔥

File: shared/devices/device-revoke/index.js [hunk 6]
```diff
@@ -1,31 +1,49 @@
...
-  </Box>
-)
+const Header = ({name, type}: {name: string, type: DeviceType}) => {
+  const icon: IconType = {
+    backup: isMobile ? 'icon-paper-key-revoke-64' : 'icon-paper-key-revoke-48',
+    desktop: isMobile ? 'icon-computer-revoke-64' : 'icon-computer-revoke-48',
+    mobile: isMobile ? 'icon-phone-revoke-64' : 'icon-phone-revoke-48',
+  }[type]
+  return (
+    <Box style={{...globalStyles.flexBoxColumn, alignItems: 'center'}}>
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: should move to styles

File: shared/devices/device-revoke/index.js (hunk 2 above)
Comment (nathunsmitty) [reply shared/devices/device-revoke/index.js]: Oh actually it looks like `overflowY` is not a available react native value and `overflow` has [some limitations](https://facebook.github.io/react-native/docs/layout-props.html#overflow). Anyone have any suggestions on how to get the type of scrolling behavior we get on desktop on mobile also then?

Comment (chrisnojima) [reply shared/devices/device-revoke/index.js]: just use a <ScrollView> from common adapters, you can't style overflow in RN like desktop

File: shared/devices/device-revoke/index.js (hunk 6 above)
Comment (nathunsmitty) [reply shared/devices/device-revoke/index.js]: Generally, are there any cases where styles should be defined inline like this or are we moving to the practice of putting all styles in the `styles` object?

Comment (chrisnojima) [reply shared/devices/device-revoke/index.js]: ideally they'd all be in there. plus you already have a styleSheetCreate so might as well put it in

**Comment by nathunsmitty** [comment]
@keybase/react-hackers bumping this

**Comment by nathunsmitty** [comment]
@keybase/react-hackers go for another look over!

File: shared/common-adapters/list.desktop.js [hunk 7]
```diff
@@ -36,21 +36,9 @@ class List extends PureComponent<Props<any>, void> {
...
-        <div style={globalStyles.fillAbsolute}>
-          <div
-            style={{
-              height: '100%',
-              overflowY: 'auto',
-              width: '100%',
-            }}
-          >
+      <div style={{flexGrow: 1, position: 'relative', ...this.props.style}}>
+        <div style={collapseStyles([globalStyles.fillAbsolute, {padding: 'inherit'}])}>
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/common-adapters/list.desktop.js]: lets discuss

File: shared/common-adapters/list.native.js [hunk 8]
```diff
@@ -30,13 +30,7 @@ class List extends PureComponent<Props<any>, void> {
...
   render() {
     return (
-      <View
-        style={{
-          flexGrow: 1,
-          position: 'relative',
-          ...this.props.style,
-        }}
-      >
+      <View style={collapseStyles([styles.outerView, this.props.style])}>
```
Comment (nathunsmitty) [inline shared/common-adapters/list.native.js]: I was getting errors with spreading `this.props.style` on mobile, so it seemed like a good opportunity to do some minor refactoring using `collapseStyles` and `styleSheetCreate`.

File: shared/devices/device-revoke/index.js [hunk 9]
```diff
@@ -71,33 +100,56 @@ const DeviceRevoke = (props: Props) => (
...
+      borderWidth: 1,
+      marginBottom: globalMargins.small,
+      marginTop: globalMargins.small,
+    },
+    isElectron: {
+      height: 162,
+      width: 440,
+    },
+    isMobile: {
+      height: 176,
```
Comment (nathunsmitty) [inline shared/devices/device-revoke/index.js]: The endangeredTLFs list is currently kinda buggy and I think height was set sort of implicitly before, so I decided to just set a list height of 176, which looks pretty good on all phone sizes.

File: shared/common-adapters/list.native.js (hunk 8 above)
Comment (chrisnojima) [reply shared/common-adapters/list.native.js]: 👍 we definitely want all the common-adapters to be able to take ids

File: shared/devices/device-revoke/index.js (hunk 9 above)
Comment (chrisnojima) [reply shared/devices/device-revoke/index.js]: can you show me this

**Comment by chrisnojima** [comment]
also ci isn't passing

File: shared/devices/device-revoke/index.js [hunk 10]
```diff
@@ -1,103 +1,195 @@
...
-              </Text>
-              <Text type="BodySemibold">{tlf}</Text>
-            </Box>
-          ))}
-        </Box>
-      </Box>
-    )}
-  </Box>
+const ActionButtons = (props: {onCancel: () => void, onSubmit: () => void, waiting: boolean}) => (
+  <Box2 direction={isMobile ? 'vertical' : 'horizontalReverse'} style={styles.actionButtonsContainer}>
```
Comment (nathunsmitty) [inline shared/devices/device-revoke/index.js]: This is the reason why I added `horizontalReverse` and `verticalReverse` – gives us the delete button on top in mobile and to the right on desktop!

**Comment by nathunsmitty** [comment]
@chrisnojima I know you approved, but may want to take a look at the row-reverse Box2 stuff

File: shared/common-adapters/box.desktop.js [hunk 11]
```diff
@@ -27,9 +27,27 @@ const injectGaps = (Component, _children, gap, gapStart, gapEnd) => {
...
   render() {
-    const horizontal = this.props.direction === 'horizontal'
+    let directionStyle
+    let horizontal = this.props.direction === 'horizontal' || this.props.direction === 'horizontalReverse'
+    switch (this.props.direction) {
+      case 'horizontal':
+        directionStyle = globalStyles.flexBoxRow
+        break
+      case 'horizontalReverse':
+        directionStyle = {display: 'flex', flexDirection: 'row-reverse'}
```
Comment (chrisnojima) [inline shared/common-adapters/box.desktop.js]: this has to pull from styles, not make a new object every time or you'll re render over and over

File: shared/common-adapters/box.desktop.js [hunk 12]
```diff
@@ -62,13 +80,7 @@ const styles = {
...
   fullHeight: {height: '100%'},
   fullWidth: {width: '100%'},
-  hbox: {
-    ...globalStyles.flexBoxRow,
-    alignItems: 'stretch',
-    justifyContent: 'flex-start',
-  },
-  vbox: {
-    ...globalStyles.flexBoxColumn,
+  commonDirectionStyles: {
```
Comment (chrisnojima) [inline shared/common-adapters/box.desktop.js]: this isn't supposed to be common, its different vertical / horizontal

File: shared/common-adapters/box.desktop.js [hunk 13]
```diff
@@ -27,9 +27,27 @@ const injectGaps = (Component, _children, gap, gapStart, gapEnd) => {
...
+    let directionStyle
+    let horizontal = this.props.direction === 'horizontal' || this.props.direction === 'horizontalReverse'
+    switch (this.props.direction) {
+      case 'horizontal':
+        directionStyle = globalStyles.flexBoxRow
+        break
+      case 'horizontalReverse':
+        directionStyle = {display: 'flex', flexDirection: 'row-reverse'}
+        break
+      case 'verticalReverse':
```
Comment (chrisnojima) [inline shared/common-adapters/box.desktop.js]: it should just map from the 4 values to 4 things in the stylesheet create

**Comment by chrisnojima** [comment]
need to add these cases to box.stories.js

**Comment by nathunsmitty** [comment]
@keybase/react-hackers this could use some 👀

File: shared/devices/device-revoke/index.js [hunk 14]
```diff
@@ -1,103 +1,195 @@
...
+    <Button
+      fullWidth={!!isMobile}
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: can jsut be isMobile, its already a boolean

File: shared/devices/device-revoke/index.js [hunk 15]
```diff
@@ -1,103 +1,195 @@
...
+      type="Danger"
+      onClick={props.waiting ? null : props.onSubmit}
+      label="Yes, delete it"
+      disabled={!!props.waiting}
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: same here

File: shared/devices/device-revoke/index.js [hunk 16]
```diff
@@ -1,103 +1,195 @@
...
+      style={isMobile ? {marginBottom: globalMargins.tiny} : {marginLeft: globalMargins.tiny}}
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: cant you just use gapEnd in the Box2 parent

File: shared/devices/device-revoke/index.js [hunk 17]
```diff
@@ -1,103 +1,195 @@
...
-    onSubmit={props.waiting ? null : props.onSubmit}
-    disabled={!!props.waiting}
-    submitLabel="Yes, delete it"
-    theme="public"
-  />
+  <Box2 direction="vertical" fullHeight={true} fullWidth={true}>
+    <Box style={styles.deviceRevokeContainer}>
+      <Header name={props.name} type={props.type} />
+      <BodyText name={props.name} currentDevice={props.currentDevice} />
+      <EndangeredTLFList endangeredTLFs={props.endangeredTLFs} waiting={props.waiting} style={{flex: 1}} />
```
Comment (chrisnojima) [inline shared/devices/device-revoke/index.js]: use styles below

**Comment by nathunsmitty** [comment]
@chrisnojima made your fixes!

**Pull request merged**